core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -185,12 +186,59 @@ public void cancelOperation() throws UnsupportedOperationException {
      * This method returns a {@link Flux} that can be subscribed to, enabling a subscriber to receive notification of
      * every {@link PollResponse}, as it is received.
      *
+     * This will return updated {@link Flux} if user had made call to {@link Poller#getObserver(List, List)} earlier and will selectively receive notification for
+     * only those {@link com.azure.core.util.polling.OperationStatus.State} specified in {@link Poller#getObserver(List, List)}.
+     *
      * @return A {@link Flux} that can be subscribed to receive poll responses as the long-running operation executes.
      */
     public Flux<PollResponse<T>> getObserver() {
         return this.fluxHandle;
     }
 
+    /**
+     * This method returns an updated {@link Flux} that is observing specified {@link com.azure.core.util.polling.OperationStatus.State} and
+     * can be subscribed to, enabling a subscriber to receive these specific notification of {@link PollResponse}, as it is received.
+     * @param observeState which user want to observe, must specify complete {@link com.azure.core.util.polling.OperationStatus.State} if interested.
+     * User not required to specify {@link com.azure.core.util.polling.OperationStatus.State#OTHER} here if interested in other state.
+     * @param observeOtherStates which user is interested to observe.
+     * @return A {@link Flux} that can be subscribed to receive poll responses as the long-running operation executes.
+     */
+    public Flux<PollResponse<T>> getObserver(List<OperationStatus.State> observeState, List<String> observeOtherStates) {
+        if (observeState == null && (observeOtherStates == null || observeOtherStates.size() == 0)) {
+            throw new IllegalArgumentException("observeOperationStates and observeStates both can not be null or empty.");
+        }
+        //TODO : Design discussion, We may decide to add complete state. If user did not added them, we may never listen them and LRO would never complete.
+        //observeState.add(OperationStatus.State.SUCCESSFULLY_COMPLETED);
+        //observeState.add(OperationStatus.State.USER_CANCELLED);
+        //observeState.add(OperationStatus.State.FAILED);
+        this.fluxHandle = this.fluxHandle.filterWhen(tPollResponse -> matchesState(tPollResponse, observeState, observeOtherStates));
+        return this.fluxHandle;
+    }
```

Comment:
I don't think this API is necessary. The user can implement the filter on their side.

Reply:
That's true , makes poller simpler.